### ISSUE TITLE
Pod cannot run on K8s versions lower than 1.29.

### ISSUE THREAD

**Issue by ty-dc** [spidernet-io/spiderpool#3180]
### Spiderpool Version

v0.9.0

### Bug Type

Coordinator

### Main CNI

macvlan

### What happened?

```
  Warning  FailedCreatePodSandBox  5m6s  kubelet  Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "e16bca5abd9e5a492ca7d292c452338e355f5769a736342112e4b2969790d2e8": plugin type="multus" name="multus-cni-network" failed (add): [kruise-system/kruise-controller-manager-5b87dd5d49-8tqcw/3c09d2cb-359f-43f3-b08a-e1e109c5c0fe:macvlan-vlan0]: error adding container to network "macvlan-vlan0": validation failure list:
serviceCIDR in body is required
  Warning  FailedCreatePodSandBox  48s (x17 over 4m50s)  kubelet  (combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox 
```

<img width="829" alt="企业微信截图_f97c01db-d9bb-4e64-b946-41ac9c39e671" src="https://github.com/spidernet-io/spiderpool/assets/101157744/89f5ecbe-d2ab-4527-835e-f6d502bcd46a">

### What did you expect to happen?

The pod runs successfully.

### How to reproduce it (as minimally and precisely as possible)

make e2e_init_underlay  -e E2E_CHINA_IMAGE_REGISTRY=true -e E2E_KIND_IMAGE_TAG=v1.23.5 -e  INSTALL_KRUISE=true

**Comment by ty-dc** [comment]
fix #3168

**Comment by ty-dc** [comment]
> @ty-dc - When will this fix be released, I am experiencing the same bug currently

What version did you use to encounter this problem?

**Comment by infinitydon** [comment]
I used the 0.91 chart:

```
spiderpool                              kube-system     1               2024-02-05 15:32:44.330109507 +0000 UTC deployed        spiderpool-0.9.1                                0.9.1
```

**Comment by ty-dc** [comment]
Thanks, I tried version 0.9.1 and it doesn't seem to be a problem? Can you provide more information?

```
[root@controller-node-1 ~]# helm list -A
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART               APP VERSION
spiderpool      kube-system     1               2024-02-04 13:57:52.70865064 +0800 CST  deployed        spiderpool-0.9.1    0.9.1      

[root@controller-node-1 ~]# kubectl get nodes -owide
NAME                STATUS   ROLES           AGE   VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE                KERNEL-VERSION               CONTAINER-RUNTIME
controller-node-1   Ready    control-plane   8d    v1.27.5   10.7.168.71   <none>        CentOS Linux 7 (Core)   3.10.0-957.27.2.el7.x86_64   containerd://1.7.11
worker-node-1       Ready    <none>          8d    v1.27.5   10.7.168.72   <none>        CentOS Linux 7 (Core)   3.10.0-957.27.2.el7.x86_64   containerd://1.7.11

[root@controller-node-1 ~]# kubectl get spidercoordinators.spiderpool.spidernet.io default -oyaml
apiVersion: spiderpool.spidernet.io/v2beta1
kind: SpiderCoordinator
metadata:
  creationTimestamp: "2024-02-04T06:11:53Z"
  finalizers:
  - spiderpool.spidernet.io
  generation: 1
  name: default
  resourceVersion: "1568865"
  uid: ad42115a-c181-416a-b982-2179befe9786
spec:
  detectGateway: false
  detectIPConflict: false
  hijackCIDR:
  - 169.254.0.0/16
  hostRPFilter: 0
  hostRuleTable: 500
  mode: auto
  podCIDRType: auto
  podDefaultRouteNIC: ""
  podMACPrefix: ""
  tunePodRoutes: true
  txQueueLen: 0
status:
  overlayPodCIDR:
  - 10.233.64.0/18
  - fd85:ee78:d8a6:8607::1:0/112
  phase: Synced
  serviceCIDR:
  - 10.233.0.0/18
  - fd85:ee78:d8a6:8607::1000/116
  - 
[root@controller-node-1 ~]# kubectl get network-attachment-definitions.k8s.cni.cncf.io -n kube-system macvlan-ens192 -oyaml
apiVersion: k8s.cni.cncf.io/v1
kind: NetworkAttachmentDefinition
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"spiderpool.spidernet.io/v2beta1","kind":"SpiderMultusConfig","metadata":{"annotations":{},"name":"macvlan-ens192","namespace":"kube-system"},"spec":{"cniType":"macvlan","enableCoordinator":true,"macvlan":{"master":["ens192"]}}}
  creationTimestamp: "2024-02-06T14:15:27Z"
  generation: 1
  name: macvlan-ens192
  namespace: kube-system
  ownerReferences:
  - apiVersion: spiderpool.spidernet.io/v2beta1
    blockOwnerDeletion: true
    controller: true
    kind: SpiderMultusConfig
    name: macvlan-ens192
    uid: dec250d4-a1d5-439e-90cc-3ce875d5c661
  resourceVersion: "1569818"
  uid: 8d25a549-3697-4df3-a069-a5042ceeede5
spec:
  config: '{"cniVersion":"0.3.1","name":"macvlan-ens192","plugins":[{"type":"macvlan","master":"ens192","mode":"bridge","ipam":{"type":"spiderpool"}},{"detectIPConflict":false,"detectGateway":false,"mode":"auto","type":"coordinator"}]}'

[root@controller-node-1 ~]# kubectl get po -owide
NAME                        READY   STATUS    RESTARTS   AGE     IP             NODE                NOMINATED NODE   READINESS GATES
test-app-854d7dccdc-4vgsv   1/1     Running   0          3m43s   10.6.168.201   controller-node-1   <none>           <none>
test-app-854d7dccdc-jlvvl   1/1     Running   0          3m43s   10.6.168.210   worker-node-1       <none>           <none>
```

Maybe I missed some details, can you add them? Steps similar to the following?

```
1. kubectl get spidercoordinators.spiderpool.spidernet.io default -oyaml
2. Use `kubectl describe` to view information about failed Pods
3. Logs about spiderpool-controller Pod
```

**Comment by infinitydon** [comment]
Find below for the logs:

```
kubectl get no -o wide
NAME               STATUS   ROLES                       AGE   VERSION           INTERNAL-IP      EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
o5gs-nephio-mgmt   Ready    control-plane,etcd,master   23d   v1.26.12+rke2r1   192.168.100.56   <none>        Ubuntu 20.04.6 LTS   5.4.0-170-generic   containerd://1.7.11-k3s2


 kubectl get spidercoordinators.spiderpool.spidernet.io default -oyaml
apiVersion: spiderpool.spidernet.io/v2beta1
kind: SpiderCoordinator
metadata:
  creationTimestamp: "2024-02-05T15:33:30Z"
  finalizers:
  - spiderpool.spidernet.io
  generation: 1
  name: default
  resourceVersion: "14353542"
  uid: 4c7e689b-3e4b-4d50-8bb9-6c2c391f6bb2
spec:
  detectGateway: false
  detectIPConflict: false
  hijackCIDR:
  - 169.254.0.0/16
  hostRPFilter: 0
  hostRuleTable: 500
  mode: auto
  podCIDRType: auto
  podDefaultRouteNIC: ""
  podMACPrefix: ""
  tunePodRoutes: true
  txQueueLen: 0
status:
  overlayPodCIDR:
  - 172.16.0.0/16
  phase: Synced


kubectl get network-attachment-definitions.k8s.cni.cncf.io -n kube-system macvlan-conf -oyaml
apiVersion: k8s.cni.cncf.io/v1
kind: NetworkAttachmentDefinition
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"spiderpool.spidernet.io/v2beta1","kind":"SpiderMultusConfig","metadata":{"annotations":{},"name":"macvlan-conf","namespace":"kube-system"},"spec":{"cniType":"macvlan","enableCoordinator":true,"macvlan":{"master":["ens19"]}}}
  creationTimestamp: "2024-02-06T14:37:19Z"
  generation: 1
  name: macvlan-conf
  namespace: kube-system
  ownerReferences:
  - apiVersion: spiderpool.spidernet.io/v2beta1
    blockOwnerDeletion: true
    controller: true
    kind: SpiderMultusConfig
    name: macvlan-conf
    uid: 03dfe6d8-e9eb-41d1-8cb1-6d6ab0244f09
  resourceVersion: "15247183"
  uid: 03851de7-0278-4c24-83e1-66d69045e0fc
spec:
  config: '{"cniVersion":"0.3.1","name":"macvlan-conf","plugins":[{"type":"macvlan","master":"ens19","mode":"bridge","ipam":{"type":"spiderpool"}},{"detectIPConflict":false,"detectGateway":false,"mode":"auto","type":"coordinator"}]}'


kubectl get po -owide
NAME                        READY   STATUS              RESTARTS      AGE     IP           NODE               NOMINATED NODE   READINESS GATES
test-app-578f7776d5-7rqjz   0/1     ContainerCreating   0             2m36s   <none>       o5gs-nephio-mgmt   <none>           <none>
test-app-578f7776d5-jszp5   0/1     ContainerCreating   0             2m36s   <none>       o5gs-nephio-mgmt   <none>           <none>


kubectl describe po test-app-578f7776d5-jszp5
Name:             test-app-578f7776d5-jszp5
Namespace:        default
Priority:         0
Service Account:  default
Node:             o5gs-nephio-mgmt/192.168.100.56
Start Time:       Tue, 06 Feb 2024 14:38:03 +0000
Labels:           app=test-app
                  pod-template-hash=578f7776d5
Annotations:      ipam.spidernet.io/ippool:
                    {
                      "ipv4": ["ippool-test"]
                    }
                  v1.multus-cni.io/default-network: kube-system/macvlan-conf
Status:           Pending
IP:
IPs:              <none>
Controlled By:    ReplicaSet/test-app-578f7776d5
Containers:
  test-app:
    Container ID:
    Image:          nginx
    Image ID:
    Port:           80/TCP
    Host Port:      0/TCP
    State:          Waiting
      Reason:       ContainerCreating
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-dlvlg (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  kube-api-access-dlvlg:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason                  Age   From               Message
  ----     ------                  ----  ----               -------
  Normal   Scheduled               13s   default-scheduler  Successfully assigned default/test-app-578f7776d5-jszp5 to o5gs-nephio-mgmt
  Warning  FailedCreatePodSandBox  10s   kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "2a3576fa9b27dabc9f579cf5019a4cdfea3998334c87c96d487964be7ff65c09": plugin type="multus" name="multus-cni-network" failed (add): [default/test-app-578f7776d5-jszp5/8625ffef-bc88-4316-b7b4-9a70507a44f4:macvlan-conf]: error adding container to network "macvlan-conf": validation failure list:
serviceCIDR in body is required
  Normal  SandboxChanged  10s  kubelet  Pod sandbox changed, it will be killed and re-created.
```

Error logs is saying the same thing that is mentioned in this issue

```
error adding container to network "macvlan-conf": validation failure list:
serviceCIDR in body is required
```

**Comment by infinitydon** [comment]
I followed the example manifests that was provided in the docs folder:

```
MACVLAN_MASTER_INTERFACE="ens19"
cat <<EOF | kubectl apply -f -
apiVersion: spiderpool.spidernet.io/v2beta1
kind: SpiderMultusConfig
metadata:
  name: macvlan-conf
  namespace: kube-system
spec:
  cniType: macvlan
  enableCoordinator: true
  macvlan:
   master:
   - ${MACVLAN_MASTER_INTERFACE}
EOF

cat <<EOF | kubectl apply -f -
apiVersion: spiderpool.spidernet.io/v2beta1
kind: SpiderIPPool
metadata:
  name: ippool-test
spec:
  ips:
  - "192.168.200.2-192.168.200.4"
  subnet: 192.168.200.0/24
  gateway: 192.168.200.1
  multusName: 
  - kube-system/macvlan-conf
EOF



cat <<EOF | kubectl apply -f -
apiVersion: apps/v1
kind: Deployment
metadata:
  name: test-app
spec:
  replicas: 2
  selector:
    matchLabels:
      app: test-app
  template:
    metadata:
      annotations:
        ipam.spidernet.io/ippool: |-
          {
            "ipv4": ["ippool-test"]
          }
        v1.multus-cni.io/default-network: kube-system/macvlan-conf
      labels:
        app: test-app
    spec:
      containers:
      - name: test-app
        image: nginx
        imagePullPolicy: IfNotPresent
        ports:
        - name: http
          containerPort: 80
          protocol: TCP
---
apiVersion: v1
kind: Service
metadata:
  name: test-app-svc
  labels:
    app: test-app
spec:
  type: LoadBalancer
  ports:
    - port: 80
      protocol: TCP
      targetPort: 80
  selector:
    app: test-app 
EOF
```

**Comment by ty-dc** [comment]
Did you also install the cluster through `k0s installer`?

**Comment by infinitydon** [comment]
No, cluster was created using RKE2

**Comment by ty-dc** [comment]
Thanks, I tried it on RKE2 cluster and it looks like it's a bug.

**Comment by ty-dc** [comment]
Submitted a new issue #3210 to track, thanks.

**Comment by ty-dc** [comment]
I fixed it in pr #3211.

In addition, it should be noted that there are no kube-controller-manager and kubeadm-config in the rke2 cluster. For configuration, please refer to [Document](https://github.com/spidernet-io/spiderpool/blob/main/docs/usage/install/overlay/get-started-calico.md)

Finally you will see the effect below.
```
[root@test-ss ~]# kubectl get spidercoordinators.spiderpool.spidernet.io default -oyaml
apiVersion: spiderpool.spidernet.io/v2beta1
kind: SpiderCoordinator
metadata:
  creationTimestamp: "2024-02-07T10:15:44Z"
  finalizers:
  - spiderpool.spidernet.io
  generation: 1
  name: default
  resourceVersion: "56279"
  uid: c36f765b-79f9-4475-a803-1d96213af068
spec:
  detectGateway: false
  detectIPConflict: false
  hijackCIDR:
  - 169.254.0.0/16
  hostRPFilter: 0
  hostRuleTable: 500
  mode: auto
  podCIDRType: auto
  podDefaultRouteNIC: ""
  podMACPrefix: ""
  tunePodRoutes: true
  txQueueLen: 0
status:
  overlayPodCIDR:
  - 192.168.0.0/16
  phase: Synced
  serviceCIDR:
  - 10.233.0.0/18
```
